File: samples/cpp/enroll.cc
// License: Apache 2.0. See LICENSE file in root directory.
// Copyright(c) 2020-2021 Intel Corporation. All Rights Reserved.

#include "RealSenseID/FaceAuthenticator.h"
#include <iostream>
#include <unistd.h>
#include <string.h>

void showHelp(char *_run)
{     
      std::cout<<"Usage : " << _run << " -u [UserID]" << std::endl;
}

class MyEnrollClbk : public RealSenseID::EnrollmentCallback
{
public:
    void OnResult(const RealSenseID::EnrollStatus status) override
    {
        std::cout << "Result " << status << std::endl;
    }

    void OnProgress(const RealSenseID::FacePose pose) override
    {
        std::cout << "OnProgress " << pose << std::endl;
    }

    void OnHint(const RealSenseID::EnrollStatus hint) override
    {
        std::cout << "Hint " << hint << std::endl;
    }

    void OnFaceDetected(const std::vector<RealSenseID::FaceRect>& faces) override
    {
        for (auto& face : faces)
        {
            std::cout << "Detected face " << face.x << "," << face.y << ", " << face.w << "x" << face.h << std::endl;
        }
    }
};

int main(int argc, char **argv)
{
    if(argc<2)
    {
        showHelp(argv[0]);
        exit(0);
    }
    int c;
    const char *user_id = NULL;
    RealSenseID::FaceAuthenticator authenticator;
#ifdef _WIN32
    auto status = authenticator.Connect({"COM9"});
#elif LINUX
    auto status = authenticator.Connect({"/dev/ttyACM0"});
#endif
    while ((c = getopt(argc, argv, "u:")) != -1)
    {
        switch(c)
        {
            case 'u':
            optind--;
            user_id = strdup(argv[optind]);
            break;
        }
    }

    if(status != RealSenseID::Status::Ok)
    {
        std::cout << "Failed connecting with status " << status << std::endl;
        return 1;
    }
    MyEnrollClbk enroll_clbk;
    authenticator.Enroll(enroll_clbk, user_id);
}

File: samples/python/enroll.py
"""
License: Apache 2.0. See LICENSE file in root directory.
Copyright(c) 2020-2021 Intel Corporation. All Rights Reserved.
"""
import rsid_py

PORT='COM9'

def on_result(result):
    print('on_result', result)    

def on_progress(p):    
    print(f'on_progress {p}')

def on_hint(h):    
    print(f'on_hint {h}')

def on_faces(faces, timestamp):    
    print(f'detected {len(faces)} face(s)')
    for f in faces:
        print(f'\tface {f.x},{f.y} {f.w}x{f.h}')    


if __name__ == '__main__':
    with rsid_py.FaceAuthenticator(PORT) as f:
        user_id = input("User id to enroll: ")
        f.enroll(user_id=user_id, on_hint=on_hint, on_progress=on_progress, on_faces=on_faces, on_result=on_result)

        #display list of enrolled users
        users = f.query_user_ids()
        print('Users: ', users)
    
    